src/waypoint.jsx

```diff
@@ -188,11 +192,17 @@ export default class Waypoint extends React.Component {
         currentPosition: POSITIONS.inside,
         previousPosition,
         event,
+        waypointTop: bounds.waypointTop,
+        viewportTop: bounds.viewportTop,
+        viewportBottom: bounds.viewportBottom,
       });
       this.props.onLeave.call(this, {
         currentPosition,
         previousPosition: POSITIONS.inside,
         event,
+        waypointTop: bounds.waypointTop,
+        viewportTop: bounds.viewportTop,
+        viewportBottom: bounds.viewportBottom,
```

Comment:
I think you could use an object spread here (and in two places above) to avoid having to repeat the bounds properties:

```
this.props.onLeave.call(this, {
  currentPosition,
  previousPosition: POSITIONS.inside,
  event,
  ...bounds,
});
```